sno/db_util.py

```diff
@@ -0,0 +1,61 @@
+from psycopg2.sql import Identifier, SQL
```

Comment:
this whole file feels wrong to me, feels like it's going to grow unholy quickly. Should we look into using a proper multi-database abstraction like sqlalchemy? 

We probably don't want or need an ORM, just the SQL implementation abstraction: https://docs.sqlalchemy.org/en/13/core/tutorial.html . Couple with some [table reflection](https://docs.sqlalchemy.org/en/13/core/reflection.html) for auto-creating the Table instances for user-defined tables)

One minor thing to note with using sqlalchemy is that it doesn't work with the apsw driver (rogerbinns/apsw#107) but I'm not clear if there's any reason we're using apsw to begin with instead of the sqlite module in the stdlib? (cc @rcoup) Feels like switching to something more DBAPI-compliant would make interaction with multiple databases simpler even if we didn't use sqlalchemy.

Reply:
I mentioned this in https://github.com/koordinates/sno/pull/270#issuecomment-703492688 too.

>  I'm not clear if there's any reason we're using apsw to begin with instead of the sqlite module in the stdlib?

1. The sqlite module in stdlib doesn't support loading extensions in platform/version-specific ways.
2. I wanted consistency on sqlite versions across python+spatialite+gdal+proj
3. pysqlite has been abandoned as a standalone project
4. So we'd need to build _Python_ per-platform to get the right build options and an up to date sqlite version
5. And that seemed much harder than swapping to apsw
6. pysqlite's transaction behaviour is abominable ⛄

Reply:
> Should we look into using a proper multi-database abstraction like sqlalchemy?

No objection in principle, seems better than hand-rolling string-substitution & type functions forever & ever

Reply:
aha, there is a https://github.com/coleifer/pysqlite3 now!

Reply:
and https://github.com/hoehrmann/sqlalchemy-pysqlite3

Reply:
Discussed offline - punted until after next week